### PR TITLE
Modifications to account for Connexity API

### PR DESCRIPTION
**Pull request by akatz000**
- New API endpoint
- New response structure
- New search param interactions

[#127486781]

### CODE REVIEW

File: lib/pricegrabber/request.rb [hunk 1]
```diff
@@ -66,39 +60,28 @@ def version
     end
 
     def pid
```
Comment (craineum) [inline lib/pricegrabber/request.rb]: This method could probably go away, unless it is being used from somewhere else?

File: lib/pricegrabber/response.rb [hunk 2]
```diff
@@ -9,10 +9,10 @@ class Response
     # @param wants [Array<String>] Attributes from the response that should be made available within this response
     def initialize(response, wants)
       @status = response.code
-      resp_hash = Hash.from_xml(response.body)
-      @error = resp_hash["document"]["error"]
       @results = []
-      [resp_hash["document"]["product"]].flatten.each do |product|
+      return if @status != 200
+      resp_hash = Hash.from_xml(response.body)
+      [resp_hash["ProductResponse"]["Offers"]["Offer"]].flatten.each do |product|
```
Comment (craineum) [inline lib/pricegrabber/response.rb:15]: Looking at the diff we have locked this down to offers only. That is fine for our use, but is good reason not to try and merge it up to main code base.

**Review by craineum — COMMENTED** [review]
One minor issue and one comment. Other than that it LGTM.

File: lib/pricegrabber/request.rb (hunk 1 above)
Comment (akatz000) [reply lib/pricegrabber/request.rb]: Yeah, looks like dead code.  I think I moved this into a config var

**Pull request merged**